datafusion/physical-plan/src/filter.rs

```diff
@@ -278,10 +279,12 @@ impl ExecutionPlan for FilterExec {
         trace!("Start FilterExec::execute for partition {} of context session_id {} and task_id {:?}", partition, context.session_id(), context.task_id());
         let baseline_metrics = BaselineMetrics::new(&self.metrics, partition);
         Ok(Box::pin(FilterExecStream {
```

Comment:
the changes here show how to integrate Coalesce into Filter, which seems reasonable so far